### DISCUSSION TITLE
NDI source input or Output - maybe teleport output for LAN

### DISCUSSION THREAD

**Discussion by FilipStadler** [datarhei/restreamer#391] [Ideas]
I have a new cameras than can run NDI.
OBS can get camera NDI input and it can also do NDI output.
I can use the Mevo  in NDI mode with my OBS running on Linux.
Also some cams PTZ cams provide NDI.

Using restreamer to do NDI stream to stream to other protocol destinations would be usefull.

I am not sure if possible because NDI is not opensource like SRT/RIST but then again they have solved it with OBS,

A simular solution for output is the TELEPORT its a very low latency solution like NDI but TELEPORT is only usefull for OBS to OBS connection - but have tried on raspberrypi4 with OBS.

Anyway the restreamer is very powerfull now with the SRT options also - running on a camcorder with hdmi out and usb capture card or connected to a videoswitch like maybe rgblink mini+ or fellworld makes it cool for portable livestream setup.

Also my mevo cam app can do a rtmp from to raspberry pi and I can let it do SRT - whats cool with the mevo cam is they be used with a multicam app but this only provide rtmp out - so I use this to stream how to my own restreamer.

So this maybe a feature request.

**Comment by jstabenow** [comment]
Good idea! But at the moment, it doesn't look like we are allowed to integrate NDI into FFmpeg and pass it on. 
Maybe it would be possible via an installation script. We will clarify the situation as soon as we have enough time. 

If you want to help, look at how to share FFmpeg with NDI publicly. Without a license violation.

**Reply by FilipStadler** [reply]
Yes  NDI is maybe not possible to integrate NDI into FFmpeg but OBS use a plugin so if restreamer was able to use
https://github.com/Palakis/obs-ndi like if restreamer was a OBS so the OBS source code is available.

TELEPORT is very usefull for lowlatency but only from one OBS to another or maybe from  restreamer to OBS or OBS to restreamer and even from one restreamer to another - but compared to NDI that is used by cams.

https://github.com/fzwoch/obs-teleport

I use restreamer almost as OBS plugin :)   because OBS only have a single record and streamout option and the solution to multi distribution is limited in OBS and the need to start/stop outgoing and configure streams remotely is more streams is very usefull.
Because I also use OBS with websocket remotely.  

Now I can also have resteamer with SRT on raspberryPI4 and USB capture card/or a RGBlink videomixer and this really powerfull compared with lower latency compared to a Terredek vidiu mini that only do rtmp.

A restreamer with NDI/TELEPORT/SRT/RTMP input and maybe modified "SRT" like this https://github.com/BELABOX/srtla outfut would be very powerfull and usefull alot of places because it can improve impossible connection with Limited capasity.

Also I use the restreamer on a setup with owncast on the samme computer and in both cases the need of using loopback 127.0.0.1 would be usefull but maybe its because I needed to use docker bridge interface.

I have been looking on the https://github.com/johanels/raspberrypi-FFmpegNDI but I have also found this

https://github.com/lplassman/V4L2-to-NDI but like the obs-ndi it use the newtek SDK

Anyway I must say the new restreamer is just allready great and the best solution I have found and so nice web interface.
